src/shared/Microsoft.AzureRepos/AzureReposHostProvider.cs

```diff
@@ -549,6 +549,14 @@ private bool UseServicePrincipal(out ServicePrincipalIdentity sp)
 
             if (hasCertThumbprint)
             {
+                bool hasX5CSetting = _context.Settings.TryGetSetting(
+                    AzureDevOpsConstants.EnvironmentVariables.ServicePrincipalCertificateSendX5C,
+                    Constants.GitConfiguration.Credential.SectionName,
+                    AzureDevOpsConstants.GitConfiguration.Credential.ServicePrincipalCertificateSendX5C,
+                    out string certHasX5C);
+
+                sp.SendX5C = !hasX5CSetting || certHasX5C != "false";
```

Comment:
For this we have some helpers for parsing Git config 'bool' strings to `bool` - values like `"1"` or `"on"` should also be treated as `true`.

Taking the raw string value from settings, you can call `ToBooleanyOrDefault` which will do this conversion for you.. use `ToBooleanyOrDefault(false)` to set the default, if unset or invalid, to `false` to like below:

```suggestion
                sp.SendX5C = _context.Settings.TryGetSetting(
                    AzureDevOpsConstants.EnvironmentVariables.ServicePrincipalCertificateSendX5C,
                    Constants.GitConfiguration.Credential.SectionName,
                    AzureDevOpsConstants.GitConfiguration.Credential.ServicePrincipalCertificateSendX5C,
                    out string certHasX5CStr) && certHasX5CStr.ToBooleanyOrDefault(false);
```